pkg/odo/cli/component/create.go

```diff
@@ -878,6 +879,23 @@ func ensureAndLogProperResourceUsage(resource, resourceMin, resourceMax, resourc
 	}
 }
 
+func sparseCheckoutDir(project common.DevfileProject, zipURL, path string) error {
```

Comment:
Can we rename this function to something other than `sparseCheckoutDir`? Since it doesn't always do a sparse checkout

`checkoutProject`?